python/lsst/obs/base/ingest.py

```diff
@@ -920,6 +920,69 @@ def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Op
             self.log.info("Exposure %s:%s ingested successfully",
                           exposure.record.instrument, exposure.record.obs_id)
 
+        return refs, bad_files, n_exposures, n_exposures_failed, n_ingests_failed
+
+    @timeMethod
+    def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Optional[str] = None,
+            file_filter: Union[str, re.Pattern] = r"\.fit[s]?\b", group_files: bool = True):
+        """Ingest files into a Butler data repository.
+
+        This creates any new exposure or visit Dimension entries needed to
+        identify the ingested files, creates new Dataset entries in the
+        Registry and finally ingests the files themselves into the Datastore.
+        Any needed instrument, detector, and physical_filter Dimension entries
+        must exist in the Registry before `run` is called.
+
+        Parameters
+        ----------
+        files : iterable over `ButlerURI`, `str` or path-like objects
+            Paths to the files to be ingested.  Can refer to directories.
+            Will be made absolute if they are not already.
+        pool : `multiprocessing.Pool`, optional
+            If not `None`, a process pool with which to parallelize some
+            operations.
+        processes : `int`, optional
+            The number of processes to use.  Ignored if ``pool`` is not `None`.
+        run : `str`, optional
+            Name of a RUN-type collection to write to, overriding
+            the default derived from the instrument name.
+        file_filter : `str` or `re.Pattern`, optional
+            Pattern to use to discover files to ingest within directories.
+            The default is to search for FITS files.
```

Comment:
Will this search recursively, and if so, is this a pattern for the path relative to an entry in ``files``-interpreted-as-directories (including subdirs), or a pattern for the filename only, in whatever subdirectory it's found in?  (Either is reasonable, but that aspect of the behavior should be documented).

Reply:
Deep down it's using `os.walk` (or S3 equivalent) where the regex applies solely to the file name and not the directory.